.github/actions/csv-vulnerability-filter/src/main.ts

```diff
@@ -18,32 +18,35 @@ function parseIntInput(key: string, defaultValue: number): number {
     return defaultValue;
   }
   const value = parseInt(input, 10);
-  if (isNaN(value)) {
+  if (Number.isNaN(value)) {
     core.warning(`Invalid integer input for ${key}: "${input}". Using default ${defaultValue}.`);
     return defaultValue;
   }
   return value;
 }
 
-async function errorString(error: unknown): Promise<string> {
+function errorString(error: unknown): string {
   return error instanceof Error ? error.message : String(error);
 }
 
 async function run(): Promise<void> {
   try {
     // --- Get Inputs ---
+
     const primaryCsvPath = core.getInput('primary_csv_path', { required: true });
     const comparisonCsvPath = core.getInput('comparison_csv_path'); // Optional path
     const ignoreUnpatched = core.getBooleanInput('ignore_unpatched');
     const minSeverity = core.getInput('filter_min_severity');
-
-    core.info('--- Action Inputs ---');
-    core.info(`Primary CSV Path: ${primaryCsvPath}`);
-    core.info(`Comparison CSV Path: ${comparisonCsvPath || '(Not provided)'}`);
-    core.info(`Ignore Unpatched: ${ignoreUnpatched}`);
-    core.info("Assuming fixed CSV column names (vuln_id, severity, package_name, fixed_ver, etc.)");
-    core.info('--- Starting Action ---');
-
+    core.group('Action Inputs', async () => {
```

Comment:
I think you want to `await` this so it prints in order
```suggestion
    await core.group('Action Inputs', async () => {
```